### PR TITLE
#2149 properly handle threadData and event_floor methods

### LINKED ISSUE #2149: [Modelica Model compilation] Properly handle _event_floor modelica method

### PR DESCRIPTION
**Pull request by rosiereflo**
closes #2149

### CODE REVIEW

**Review by adriguir — APPROVED** [review]

**Comment by adriguir** [comment]
In addition we can perhaps also handle the event_integer method that works in a similar fashion. @cossartque what do you think?

**Comment by cossartque** [comment]
> In addition we can perhaps also handle the event_integer method that works in a similar fashion. @cossartque what do you think?

I agree, they are very similar and could be used for quite the same things

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py [hunk 1]
```diff
@@ -856,6 +856,18 @@ def transform_atan3_operator_evalf(line):
 
     return line_tmp_bis
 
+##
+# Transform _event_floor(x, index, data) to (modelica_integer)floor(x)
+# @param line : line to analyse
+# @return line transformed
+def replace_event_floor(line):
+    if "_event_floor" not in line:
```
Comment (cossartque) [inline dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py:864]: according to the update of the FixedBooleanDelay Model, _event_integer should also be added

**Comment by adriguir** [comment]
I would like to separate the PR in two parts and not to mix two different things. 
The PR in its original version enables to support the event_floor support and then solves the original problem written in the issue. 
Then there is something that doesn't work is the use of the keyword "delay" on a boolean, this is another issue and I'm fine with this behavior because it doesn't make sense to interpolate on boolean values. So this should be in another issue and the correction discussed and done in another PR.

Consequently, the second commit should be reverted here and a new issue should be opened for the delayBoolean where we can also discuss the implementation (I don't like the current one)

File: dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/FixedBooleanDelay.mo [hunk 2]
```diff
@@ -23,10 +23,13 @@ block FixedBooleanDelay "Delay block with fixed delay time for boolean input"
   parameter Boolean Y0 "Initial value of output";
 
 protected
+  Real uReal;
```
Comment (cossartque) [inline dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/FixedBooleanDelay.mo:26]: @ErwanGuichard49, could you revert ?

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py [hunk 3]
```diff
@@ -856,6 +856,20 @@ def transform_atan3_operator_evalf(line):
 
     return line_tmp_bis
 
+##
+# Transform _event_floor(x, index, data) to (modelica_integer)floor(x)
```
Comment (adriguir) [inline dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py:860]: Please adapt the comment for readability and maintenance purposes.

**Review by adriguir — CHANGES_REQUESTED** [review]

File: dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py (hunk 3 above)
Comment (rosiereflo) [reply dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py:860]: done

Comment (adriguir) [reply dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/utils.py:860]: Thanks

**Review by adriguir — APPROVED** [review]

**Pull request merged**